pydantic/_internal/_dataclasses.py

```diff
@@ -152,3 +156,22 @@ class B(A):
         and not hasattr(_cls, '__pydantic_validator__')
         and set(_cls.__dataclass_fields__).issuperset(set(getattr(_cls, '__annotations__', {})))
     )
+
+
+def rebuild_pydantic_dataclass(cls: type[PydanticDataclass], *, types_namespace: dict[str, Any] | None = None) -> None:
```

Comment:
I think @adriangb noticed this, but rebuilding the dataclass causes you to lose information about what was on the `dataclasses.field(...)` default value.

One solution would be to store the "original" field defaults on the class so they can be re-considered if/when rebuilding is necessary. But perhaps there is a more efficient solution?